scripts/core/helpers/url.ts

```diff
@@ -0,0 +1,103 @@
+import ng from 'core/services/ng';
+import {ISuperdesk} from 'superdesk-api';
+
+export function getUrlPage(): string {
+    const hash = window.location.hash;
+    const index = hash.indexOf('?');
+
+    return index < 0 ?
+        hash.slice(1) :
+        hash.slice(1, index);
+}
+
+export function setUrlPage(page: string): Promise<void> {
+    return ng.getService('$location').then(($location) => {
```

Comment:
is there any difference between using this and [ng.get](https://github.com/superdesk/superdesk-client-core/blob/develop/scripts/core/services/ng.ts#L27)? Apart from being async

Reply:
`getService` waits until the `$injector != null` before returning the service instance, where as `get` will throw an error if `$injector` is not defined. Considering this is changing the page, I didn't see any need to have this code run synchronously.

Reply:
Before initializing the app/extensions, we wait for angular to load, so it can be safely assumed that all code called from within react components can get services synchronously and will not throw an error.